### PR TITLE
[chrisAPI] Type declaration files and generating tools

### PR DESCRIPTION
**Pull request by zrthxn**
### Summary
- [x] Added Typescript and used it to generate typedef files. These are under `dist/types`.
- [x] Added CoreJS 3 to dependencies and changed babel target to include Node ^8.10.
- [x] Added webpack mode to "production".
- [x] Fix return types where the type is `Object`.

This is to enable this package to generate its own TS types to be used in the TS based UI. The types are generated with the `yarn build` command, no manual maintenance is needed. These are generated based on the JSDoc comments, thus those should be kept updated.

**Output of Build**

Screenshot from 2021-07-28 01-16-07

### CODE REVIEW

File: js/chrisAPI/.babelrc [hunk 1]
```diff
@@ -2,7 +2,15 @@
   "presets": [
     [
       "@babel/preset-env", {
-        "useBuiltIns": "usage"
+        "useBuiltIns": "usage",
+        "corejs": "3",
+        "targets": {
+          "node": "8.10",
+          "browsers": [
+            "last 2 versions",
```
Comment (zrthxn) [inline js/chrisAPI/.babelrc:10]: I'm only slightly apprehensive of this line. But if it means the last two major versions then we might be okay?

**Review by zrthxn — COMMENTED** [review]
Slightly apprehensive of the browser versions.

File: js/chrisAPI/.babelrc (hunk 1 above)
Comment (PintoGideon) [reply js/chrisAPI/.babelrc:10]: @zrthxn , We should be good there. Most developers set their version lists using queries like last 2 versions to be free from updating versions manually. Browserslist will use caniuse-lite with Can I Use data for this queries.

**Review by PintoGideon — APPROVED** [review]

**Review by PintoGideon — APPROVED** [review]
LGTM.

**Pull request merged**